src/XrdTpc/XrdTpcConfigure.cc

```diff
@@ -82,6 +83,14 @@ bool TPCHandler::Configure(const char *configfn, XrdOucEnv *myEnv)
             } else {
                 m_first_timeout = 2*m_timeout;
             }
+        } else if (!strcmp("http.tpc", val)) {
```

Comment:
This really creates a sore thumb here. As we always try to enforce that each component deal with only one of two prefixes: the one that is exclusive to the component and "all.". Yes, I know, there are exceptions but let's try to squash them when we can. Since this is the TPC component the exclusive prefix is "tpc." the "http." prefix is exclusive to the http component parsed in XrdHttp. So, I would strongly suggest that this really be "tpc.http" at best (you might think of some other keyword). The addition of fccreds is fine. Now I could argue that this is not specific to TPC but any component that would use curl and hence http. So, I could also argue that this really belongs in XrdHttpProtocol::Config() and it becomes a global setting for all uses of curl. That said, I'd like to see @amadio opinion on that.

Reply:
We have `ofs.tpc`, so I thought that `http.tpc` was fine, also because we already have `http.desthttps` being used in this same file. However, I agree with your arguments above. If we have `tpc.` as prefix, having many `*.tpc` options can become confusing. We could merge all authentication related options for TPC into a new `tpc.auth` or similar option that would work for both `http.tpc` and `ofs.tpc` (keeping the old style for backward compatibility). Or we could as you suggest create an `http.auth` or similar where we'd set this as `http.auth fcreds ztn`, for example. This option could then be extended to enable Kerberos authentication via [negotiate](https://everything.curl.dev/libcurl-http/auth.html#negotiate) method with something like `http.auth negotiate` if we don't already try that.